raiden/blockchain_events_handler.py

```diff
@@ -87,67 +88,28 @@ def handle_channel_new(raiden: "RaidenService", event: Event):
     is_participant1_handled_lc = LightClientService.is_handled_lc(to_checksum_address(encode_hex(participant1)),
                                                                   raiden.wal)
     is_participant2_handled_lc = LightClientService.is_handled_lc(to_checksum_address(encode_hex(participant2)),
                                                                   raiden.wal)
     is_light_channel = is_participant1_handled_lc or is_participant2_handled_lc
-    if is_participant or is_light_channel:
-        channel_proxy = raiden.chain.payment_channel(
-            canonical_identifier=CanonicalIdentifier(
-                chain_identifier=views.state_from_raiden(raiden).chain_id,
-                token_network_address=token_network_identifier,
-                channel_identifier=channel_identifier,
-            )
-        )
-        token_address = channel_proxy.token_address()
-        channel_state = get_channel_state(
-            token_address=typing.TokenAddress(token_address),
-            payment_network_identifier=raiden.default_registry.address,
-            token_network_address=token_network_identifier,
-            reveal_timeout=raiden.config["reveal_timeout"],
-            payment_channel_proxy=channel_proxy,
-            opened_block_number=block_number,
-            is_light_channel=is_light_channel,
-            both_participants_are_light_clients=is_participant1_handled_lc and is_participant2_handled_lc
-        )
-
-        if is_participant1_handled_lc or is_participant2_handled_lc:
-            if is_participant1_handled_lc:
-                if participant1 != channel_state.our_state.address:
-                    # Swap our_state and partner_state in order to have the LC from our_side of the channel
-                    channel_state.our_state, channel_state.partner_state = channel_state.partner_state, channel_state.our_state
-                # swap channel proxy to ensure proxy.participant1 is the light client
-                channel_proxy.swap_participants(participant1)
-            else:
-                if participant2 != channel_state.our_state.address:
-                    # Swap our_state and partner_state in order to have the LC from our_side of the channel
-                    channel_state.our_state, channel_state.partner_state = channel_state.partner_state, channel_state.our_state
-                # swap channel proxy to ensure proxy.participant1 is the light client
-                channel_proxy.swap_participants(participant2)
-
-        new_channel = ContractReceiveChannelNew(
-            transaction_hash=transaction_hash,
-            channel_state=channel_state,
-            block_number=block_number,
-            block_hash=block_hash,
-        )
-        raiden.handle_and_track_state_change(new_channel)
-
-        partner_address = channel_state.partner_state.address
-
-        light_client_address = None
+    if is_light_channel:
```

Comment:
this variable serves no purpose. just do:

```python
if is_p1_lc:
  create_lc(...)
if is_p2_lc:  # should this be elif instead?
  create_lc(...)
elif:
  ...
```

Reply:
It's ok as an if, as both could be light clients.

I don't think the code you suggest is actually equivalent to the original one.

Reply:
What about something like:

```suggestion
    if is_light_channel or raiden.address in (participant1, participant2):
        if raiden.address == participant1 or is_participant1_handled_lc:
            create_channel(
                block_hash,
                block_number,
                channel_identifier,
                participant1,
                participant2,
                token_network_identifier,
                transaction_hash,
                raiden
            )
        if raiden.address == participant2 or is_participant2_handled_lc:
            create_channel(
                block_hash,
                block_number,
                channel_identifier,
                participant2,
                participant1,
                token_network_identifier,
                transaction_hash,
                raiden)
    # Raiden node is not participant of channel. Lc are not participants
    else:
```